src/models/ruleSet.model.js

```diff
@@ -7,34 +7,35 @@ module.exports = class RuleSet {
     return tasks[id]
   }
 
-  static async getValidRuleSetIds (context, applicationLineId) {
-    const dynamicsDal = new DynamicsDalService(context.authToken)
-    const ruleSetIds = []
-    Object.entries(tasks)
-      .forEach(([prop, { ruleSetId }]) => {
-        if (ruleSetId) {
-          // TODO: EWC: Remove this check when bespoke tasks are implemented
-          if (ruleSetId !== 'defra_wastetypeslistrequired') {
-            ruleSetIds.push(ruleSetId)
+  static async getValidRuleSetIds (context) {
+    if (!context.validRuleIds) {
+      const { authToken, applicationLineId } = context
+      const dynamicsDal = new DynamicsDalService(authToken)
+      const ruleSetIds = []
+      Object.entries(tasks)
+        .forEach(([prop, { ruleSetId }]) => {
+          if (ruleSetId) { // TODO: EWC: Remove this check when bespoke tasks are implemented
+            if (ruleSetId !== 'defra_wastetypeslistrequired') {
+              ruleSetIds.push(ruleSetId)
```

Comment:
We should be able to clear this ruleSetId now so this check should no longer be necessary

Reply:
Removed and cleared as requested